File: pset6/caesar.py
import sys
def main():
    key = 0
    # checking the correctness of input
    if len(sys.argv) != 2: 
        print("oops :( \n")
        return 1
    else:
        key = int(sys.argv[1])
    # taking user`s text or word to be encrypted     
    s = str(input("plaintext: "))
    print("ciphertext: ", sep='', end='')
    for i in range(len(s)):
        if s[i].isalpha():
            # formula for uppercase letters. Encrypting input moving through the alphabet 
            if s[i].isupper():
                hell = (ord(s[i]) - ord('A') + key) % 26 + ord('A')
                print(chr(hell), end='')
            # formula for lowercase letters.    
            else:
                hell = (ord(s[i]) - ord('a') + key) % 26 + ord('a')
                print(chr(hell), end='')
        else:
            print(s[i], end='')
    print()
main()

File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
                        
int main(int argc, string argv[])
{
    int key = 0;
    if(argc != 2)
        {
            printf("oops :( \n");
            return 1;
        }
        else
        {
            key = atoi(argv[1]);
        }
        
    printf("plaintext: ");
    string s = get_string();
    
    printf("ciphertext: ");
    
    for(int i = 0; i < strlen(s); i++)
    {
       if(isalpha(s[i]))
        {
            if (s[i] >= 'A' && s[i] <= 'Z')
            {
             printf("%c", 'A' + (s[i] - 'A' + key) % 26);
            }
             
            else if (s[i] >= 'a' && s[i] <= 'z')
            {
            printf("%c", 'a' + (s[i] - 'a' + key) % 26);
            }
            
        }
        else
        {
        
        printf("%c", s[i]);
        }
    }
    
    printf("\n");
    return 0;  
}